packages/affine/block-embed/src/embed-linked-doc-block/adapters/plain-text.ts

```diff
@@ -0,0 +1,37 @@
+import { EmbedLinkedDocBlockSchema } from '@blocksuite/affine-model';
+import {
+  BlockPlainTextAdapterExtension,
+  type BlockPlainTextAdapterMatcher,
+  toURLSearchParams,
+} from '@blocksuite/affine-shared/adapters';
+
+export const embedLinkedDocBlockPlainTextAdapterMatcher: BlockPlainTextAdapterMatcher =
+  {
+    flavour: EmbedLinkedDocBlockSchema.model.flavour,
+    toMatch: () => false,
+    fromMatch: o => o.node.flavour === EmbedLinkedDocBlockSchema.model.flavour,
+    toBlockSnapshot: {},
+    fromBlockSnapshot: {
+      enter: (o, context) => {
+        const { configs, textBuffer } = context;
+        // Parse as link
+        if (!o.node.props.pageId) {
+          return;
+        }
+        const title = configs.get('title:' + o.node.props.pageId) ?? 'untitled';
+        const { mode, blockIds, elementIds } = o.node.props;
+        const baseUrl = configs.get('docLinkBaseUrl') ?? '';
```

Comment:
use `GenerateDocUrlProvider`, see https://github.com/toeverything/blocksuite/pulls

Reply:
These adapters may be used without BlockStd and providers.